File: python/283_move_zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        partition = 0
                
        for checkIndex in range(len(nums)):
            if nums[checkIndex] != 0:
                nums[partition], nums[checkIndex] = nums[checkIndex], nums[partition]
                partition += 1
            
        return nums
        
        

File: swift/283_move_zeroes.swift
class Solution {
  func moveZeroes(_ nums: inout [Int]) {
    var left: Int = 0

    for right in 0..<nums.count {
      if nums[right] != 0 {
        var tmp = nums[left]
        nums[left] = nums[right]
        nums[right] = tmp
        left += 1
      }
    }
  }
}
